### PR TITLE
added a user provider

### PR DESCRIPTION
**Pull request by ConnorBlakeney**
#### Changes Made
1. Added file `userprovider.js` to `users` directory.
​
#### Steps to Review
1. Checkout this branch locally and run the application.
```
git fetch --all
git checkout cb-user-provider
```
2. Run the application.
```
npm start
```
3. Open a new Terminal tab (⌘T) and navigate to the server directory.
```
..
server
```
4. Initialize virtual environment, and run the server.
```
pipenv shell
watchgod request_handler.main
```
5. Test app functionality. 
> When user clicks on `New Post`, they should be redirected to a the PostForm page
> When user enters information into the form fields and clicks `Save`, the post database should be updated with new post object and user should be redirected to the PostDetails page to view the post they just created.
6. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.

### CODE REVIEW

File: src/components/users/UserProvider.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+export const UserContext = React.createContext()
+
+/*
+ This component establishes what data can be used.
+ */
+export const UserProvider = (props) => {
+    const [users, setUsers] = useState([])
+
+    const getUsers = () => {
+        return fetch("http://localhost:8088/users")
```
Comment (samanthajmaas) [inline src/components/users/UserProvider.js:16]: needs to be localhost:8000

**Comment by ConnorBlakeney** [comment]
1) added a user directory
2)added a user provider folder in that directory
3) changed the ports in login, register, userprovider from 8088 to 8000

**Review by samanthajmaas — APPROVED** [review]

File: src/components/auth/Register.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ export const Register = (props) => {
     const firstName = useRef()
     const lastName = useRef()
     const email = useRef()
-    const bio = useRef()
+    const displayName = useRef()
```
Comment (Raparker88) [inline src/components/auth/Register.js:9]: can this be display_name so all the variables are the same as what will be in the database

**Comment by ConnorBlakeney** [comment]
changed displayName to display_name in newUser

File: src/components/auth/Register.js [hunk 3]
```diff
@@ -75,8 +77,8 @@ export const Register = (props) => {
                     <input ref={verifyPassword} type="password" name="verifyPassword" className="form-control" placeholder="Verify password" required />
                 </fieldset>
                 <fieldset>
-                    <label htmlFor="verifyPassword"> Verify Password </label>
-                    <textarea ref={bio} name="bio" className="form-control" placeholder="Let other gamers know a little bit about you..." />
+                    <label htmlFor="verifyPassword"> Choose a Display Name </label>
```
Comment (heymonicakay) [inline src/components/auth/Register.js:80]: Could you change the label 'htmlFor' to indicate that it's labeling the display name field?

**Review by heymonicakay — CHANGES_REQUESTED** [review]
Since this pull request is only intended to add the user provider, could you remove the registration and login updates and resubmit? Also, please remove the comments from the user provider file as they are not really necessary. Thank you!

**Pull request closed without merging**